server/config/config.go

```diff
@@ -735,6 +737,20 @@ func (c *Configurator) GetSAMLConfig() *SAMLConfig {
 	return &c.gc.Auth.SAMLConfig
 }
 
+func (c *Configurator) GetSelfAuthIssuer() *url.URL {
+	if !c.gc.Auth.EnableSelfAuth {
+		return nil
+	}
+	u, err := url.Parse(c.GetAppBuildBuddyURL())
+	if err != nil {
+		u = &url.URL{
+			Scheme: "http",
+			Host:   "localhost",
+		}
+	}
+	return u
+}
+
```

Comment:
nit: most of the code in configurator is just boilerplate that reads a field and returns it. Could you instead return a bool indicating if self auth is enabled and do this url construction in the place you need the url?

Reply:
Fixed.